### PR TITLE
[SPARK-49075][BUILD] Upgrade JUnit5 related to the latest version

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to upgrade `JUnit5` related to the latest version, includes:
- `junit-jupiter` from `5.9.3` to `5.11.0`.
- `junit-platform` from `1.9.3` to `1.11.0`.
- `sbt-jupiter-interface` from `0.11.1` to `0.13.0`.

### Why are the changes needed?
- `junit-jupiter` & `junit-platform`
  https://github.com/junit-team/junit5/releases/tag/r5.11.0 (JUnit 5.11.0 = Platform 1.11.0 + Jupiter 5.11.0 + Vintage 5.11.0)
  https://junit.org/junit5/docs/5.10.3/release-notes/
  https://junit.org/junit5/docs/5.10.2/release-notes/
  https://junit.org/junit5/docs/5.10.1/release-notes/
  https://junit.org/junit5/docs/5.10.0/release-notes/

- `sbt-jupiter-interface`
  Note: `sbt-jupiter-interface` 0.11.3 is published under `com.github.sbt.junit`.
  https://github.com/sbt/sbt-jupiter-interface/releases/tag/v0.13.0 (Update JUnit to 5.11.0 by @panbingkun in https://github.com/sbt/sbt-jupiter-interface/pull/98)
  https://github.com/sbt/sbt-jupiter-interface/releases/tag/v0.12.0
  https://github.com/sbt/sbt-jupiter-interface/releases/tag/v0.11.4
  https://github.com/sbt/sbt-jupiter-interface/releases/tag/v0.11.3
  image

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Review by LuciferYang — CHANGES_REQUESTED** [review]
The support for Junit 5.10.x has not been released in sbt-jupiter-interface 0.11.4 (0.11.4 only tested Junit 5.9.3). I suggest waiting for the next version of sbt-jupiter-interface to upgrade them together.

**Comment by panbingkun** [comment]
> The support for Junit 5.10.x has not been released in sbt-jupiter-interface 0.11.4 (0.11.4 only tested Junit 5.9.3). I suggest waiting for the next version of sbt-jupiter-interface to upgrade them together.

Okay, let it run `maven tests` first.

**Comment by panbingkun** [comment]
> The support for Junit 5.10.x has not been released in sbt-jupiter-interface 0.11.4 (0.11.4 only tested Junit 5.9.3). I suggest waiting for the next version of sbt-jupiter-interface to upgrade them together.

 `sbt-jupiter-interface` `0.13.9` has been released and currently supports `Junit` version `5.11.0`

**Comment by panbingkun** [comment]
After obtaining `approval`, I will remove some test code.

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -1164,3 +1164,9 @@ jobs:
           cd ui-test
           npm install --save-dev
           node --experimental-vm-modules node_modules/.bin/jest
+
+  maven-test:
```
Comment (LuciferYang) [inline .github/workflows/build_and_test.yml:1168]: please revert this change

Comment (panbingkun) [reply .github/workflows/build_and_test.yml]: Updated.

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

**Comment by LuciferYang** [comment]
image

For this, it's best for us to add a textual description in the pr description, as images are difficult to search in the git log.

**Comment by panbingkun** [comment]
> image
> For this, it's best for us to add a textual description in the pr description, as images are difficult to search in the git log.

Updated.

**Comment by LuciferYang** [comment]
Merged into master. Thanks @panbingkun
